src/libraries/System.Private.CoreLib/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Windows.cs

```diff
@@ -121,19 +121,19 @@ private static unsafe void Preallocate(string fullPath, long preallocationSize,
             {
                 int errorCode = Marshal.GetLastPInvokeError();
 
-                // we try to mimic the atomic NtCreateFile here:
-                // if preallocation fails, close the handle and delete the file
-                fileHandle.Dispose();
-                Interop.Kernel32.DeleteFile(fullPath);
-
-                switch (errorCode)
+                // Only throw for errors that indicate there is not enough space.
+                if (errorCode == Interop.Errors.ERROR_DISK_FULL ||
+                    errorCode == Interop.Errors.ERROR_FILE_TOO_LARGE)
                 {
-                    case Interop.Errors.ERROR_DISK_FULL:
-                        throw new IOException(SR.Format(SR.IO_DiskFull_Path_AllocationSize, fullPath, preallocationSize));
-                    case Interop.Errors.ERROR_FILE_TOO_LARGE:
-                        throw new IOException(SR.Format(SR.IO_FileTooLarge_Path_AllocationSize, fullPath, preallocationSize));
-                    default:
-                        throw Win32Marshal.GetExceptionForWin32Error(errorCode, fullPath);
+                    fileHandle.Dispose();
+
+                    // Delete the file we've created.
```

Comment:
This assert is on Unix but not here.
```suggestion
                    // Delete the file we've created.
                    Debug.Assert(mode == FileMode.Create || mode == FileMode.CreateNew);
```

Reply:
This method doesn't have the `mode` argument.